coins/src/utils/coins3/jsonValidation.ts

```diff
@@ -0,0 +1,61 @@
+import Ajv from "ajv";
+
+type Schema = {
+  type: string;
+  properties: { [prop: string]: { type: string } };
+  required: string[];
+  additionalProperties: Boolean;
+};
+
+const schemas: { [topic: string]: Schema } = {
+  metadata: {
+    type: "object",
+    properties: {
+      decimals: { type: "integer" },
+      symbol: { type: "string" },
+      address: { type: "string" },
+      pid: { type: "string" },
+      chain: { type: "string" },
+      source: { type: "string" },
+    },
+    required: ["decimals", "symbol", "address", "pid", "chain"],
+    additionalProperties: false,
+  },
+  current: {
+    type: "object",
+    properties: {
+      pid: { type: "string" },
+      price: { type: "number" },
+      confidence: { type: "number" },
+      source: { type: "string" },
+    },
+    required: ["pid", "price", "confidence", "source"],
+    additionalProperties: false,
+  },
+  timeseries: {
+    type: "object",
+    properties: {
+      ts: { type: "integer" },
+      pid: { type: "string" },
+      price: { type: "number" },
+      confidence: { type: "number" },
+      source: { type: "string" },
+    },
+    required: ["ts", "pid", "price", "confidence", "source"],
```

Comment:
confidence and source can be optional?

Reply:
I thought itd be a good idea to keep these as required. What would the use case be for undefined confidence or source?

Reply:
souce is not strictly necessary and missing in lot of defi adapters right? my fear is, we silently reject some tokens and wont notice it